orca-core/src/main/java/com/netflix/spinnaker/orca/DefaultStageResolver.java

```diff
@@ -61,15 +91,32 @@ public DefaultStageResolver(Collection<StageDefinitionBuilder> stageDefinitionBu
   @Override
   @Nonnull
   public StageDefinitionBuilder getStageDefinitionBuilder(@Nonnull String type, String typeAlias) {
-    StageDefinitionBuilder stageDefinitionBuilder =
-        stageDefinitionBuilderByAlias.getOrDefault(
-            type, stageDefinitionBuilderByAlias.get(typeAlias));
+    StageDefinitionBuilder stageDefinitionBuilder = getOrDefault(type, typeAlias);
 
     if (stageDefinitionBuilder == null) {
-      throw new NoSuchStageDefinitionBuilderException(
-          type, typeAlias, stageDefinitionBuilderByAlias.keySet());
+      log.debug(
+          "Stage definition builder for '{}' not found in initial stage definition builder cache, re-computing...",
+          type);
+      computeStageDefinitionBuilders();
```

Comment:
perf: Computing these every time we want a `StageDefinitionBuilder` is going to be fairly detrimental to performance given how often we're calling this. We should be able to fairly easily check for staleness so we're only creating the temp hashmap and iterating all `SDB`s when things actually change (which will more than likely only be during startup times).

Reply:
I don't know how we'd check for staleness without seeing what `StageDefintionBuilder` objects exist in the `ObjectProvider` (which we can't know unless we get all the stages from it) so I think this is necessary.  However, this will probably only be called once and AT MOST twice through the lifecycle of an application.  Same goes for the `DynamicStageResolver`.  This wont be called on every call to get a `StageDefinitionBuilder` (unless someone is looking up a stage that doesn't exist over and over again - which, looking through the logs, doesn't happen).